reconcile/utils/state.py

```diff
@@ -54,7 +54,7 @@ def ls(self):
         Returns a list of keys in the state
         """
         objects = self.client.list_objects(Bucket=self.bucket,
-                                           Prefix=self.state_path)
+                                           Prefix=f'{self.state_path}/')
```

Comment:
Is there any chance that parts of the code base rely on this less strict form as it's being used today?  In theory, they shouldn't be pulling state from other integrations, but I think it's worth verifying.

Reply:
Now that you mentioned, if someone instantiate a `State` passing `integration` as `None` or and empty string this can cause new issues.

Gonna come up with more tests.

Reply:
It'd be good to check for that.  Additionally, you might consider moving the type hints from the docstring to inline type hints.  That would also improve the likelihood of catching a `None` passed to integration (although it also wouldn't hurt to check the value of `integration`.

Reply:
I've searched through the code and only found one occasion where integration can be `''`. It is in `qontract_cli.py` as mentioned in a previous comment.

Reply:
The fix looks good to me, but I'm just wondering does the last point above mean that this will break some `qontract-cli` usage?  If so, what path would you suggest to move forward?

Reply:
To be honest, I don't know.

This method in `qontract-cli` is already "broken" because of types.

I fear that trying to fix that would make this PR too big.

Maybe I should try to fix `qontract-cli` first in another PR.

Reply:
In fact just did in #1855

Reply:
#1855 Has been merged. I'll see what does to this.

Reply:
I've update some things, including a new test to deal with `qontract-cli`